consensus/src/payload_client/mod.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::QuorumStoreError;
+use aptos_consensus_types::common::{Payload, PayloadFilter};
+use aptos_types::validator_txn::{pool::ValidatorTransactionFilter, ValidatorTransaction};
+use futures::future::BoxFuture;
+use std::time::Duration;
+
+// TODO: consider moving to a config
+pub mod mixed;
+pub mod user;
+pub mod validator;
+
+#[async_trait::async_trait]
+pub trait PayloadClient: Send + Sync {
+    async fn pull_payload(
+        &self,
+        max_poll_time: Duration,
+        max_items: u64,
+        max_bytes: u64,
+        validator_txn_filter: ValidatorTransactionFilter,
+        user_txn_filter: PayloadFilter,
+        wait_callback: BoxFuture<'static, ()>,
+        pending_ordering: bool,
+        pending_uncommitted_blocks: usize,
+        recent_max_fill_fraction: f32,
+    ) -> anyhow::Result<(Vec<ValidatorTransaction>, Payload), QuorumStoreError>;
+
+    fn trace_payloads(&self) {}
```

Comment:
is this used?

Reply:
i moved it from another file but no idea why this func is here - could not find any reference